### PR TITLE
Role delete idempotent

### PR DESCRIPTION
**Pull request by coopernurse**
Modifies `role.delete()` to check if role exists and no-ops if it does not.

I found this when testing the CloudWatch event stuff.  `kappa delete` failed when run more than once after an error.

Also cleans up some of the calls inside the `Role` class so that use of the IAM service is more consistent.

### CODE REVIEW

File: kappa/context.py [hunk 1]
```diff
@@ -177,7 +177,8 @@ def _create_event_sources(self):
             return
 
         event_source_map = {
-            'dynamodb': kappa.event_source.dynamodb_stream.DynamoDBStreamEventSource,
+            'dynamodb':
+            kappa.event_source.dynamodb_stream.DynamoDBStreamEventSource,
```
Comment (josegonzalez) [inline kappa/context.py]: Can you revert this change?

Comment (josegonzalez) [inline kappa/context.py]: Or add a skip for this pep8 violation.

**Comment by josegonzalez** [comment]
Maybe worth adding some sort of test for this?

File: kappa/context.py (hunk 1 above)
Comment (coopernurse) [inline kappa/context.py]: I reverted that line

**Comment by coopernurse** [comment]
Happy to add tests, although I don't see much existing test infrastructure here.  Would you suggest augmenting the `test_deploy.py`?  Or testing `role.py` with mocks?  Or something else?

I'm not familiar with the placebo tool that's used by `test_deploy.py` but I can dig in and figure it out if needed.

**Comment by josegonzalez** [comment]
Testing role.py with mocks works for me :)

**Comment by coopernurse** [comment]
Great - I have some mock tests ready, but it looks like the placebo tests in `tests/unit/responses/deploy/` are blowing up because `list_roles` is no longer called.

Do you know the proper procedure for maintaining that stuff?  Do we re-record them somehow?

**Comment by josegonzalez** [comment]
If we don't call it anymore, remove that part of the test.

**Comment by coopernurse** [comment]
I'm looking at this test:

https://github.com/garnaat/kappa/blob/develop/tests/unit/test_deploy.py#L46

It uses some recorded JSON files and plays them back (faking out the AWS endpoint, it appears).  So I'm not sure how one preps new JSON responses when the call sequence changes.  I'm guessing this is what the kappa `--record-path` flag is for.

I was able to get things to pass by having `_get_role()` guard on the shape of the response:

```python
    def _get_role(self):
        try:
            response = self._iam_client.call('get_role', RoleName=self.name)
            if response and 'Role' in response:
                response = response['Role']
            return response
```

But that's sort of wacky.  Since that worked it makes me wonder what these tests are really verifying.

**Comment by coopernurse** [comment]
I've pushed the unit tests (which I'm happy with), and a separate commit to `role.py` that doesn't change the behavior of the function when called against AWS but causes the `test_deploy.py` test to pass.

I'm not happy with that commit but pushed it just so you could see the difference.  I don't think it's terrible to merge that change, but I think a better solution is to re-record the files that drive `test_deploy.py`, or perhaps reconsider that test approach entirely.  The fact that my patch causes this test to work is surprising to me, as the test was recorded against a different set of AWS calls.

**Comment by coopernurse** [comment]
Looks like my random string function only works on 2.x - I'll fix that.
